cpp/inc/bond/core/schema.h

```diff
@@ -189,7 +200,7 @@ class InitSchemaDef
     template <typename T>
     bool Base(const T& /*value*/) const
     {
-        TypeDef type = GetTypeDef<T>();
+        TypeDef type = GetTypeDef<typename remove_bonded<T>::type>();
```

Comment:
When is `Base` invoked with a `bonded<>`?

In IDL, `struct Foo : bonded<Bar>` is illegal.

Reply:
Originally, `T` used to come as the base struct itself, since the transform was performed over an object. While now structs are coming as `bonded<T, Reader>` (see the `StaticParser`). That is the reason why `remove_bonded` applies recursively (at most twice), because in cases when the field is a `bonded<T>`, then the transform's `Field` will receive `bonded<bonded<T>>`. In case of a `Base`, that won't happen.